### ISSUE TITLE
Inline toolbar keyboard navigation should be consistent with block toolbar

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#43564]
### Description

Currently, the block toolbar, once focused, is navigable with the Arrow keys, because it uses the `NavigableToolbar` component. 

But the inline toolbar, such as appears for Image block captions, doesn't use the same component, so it can only be navigated with Tab key. Ideally, keyboard behaviour should be consistent across both toolbars. We could use `NavigableToolbar` for the inline toolbar too.

### Step-by-step reproduction instructions

1. Add an Image block to the page, and insert an image in it;
2. Caption should be focused automatically;
3. Press Shift+Tab to enter the caption toolbar;
4. Note that the toolbar buttons can only be reached by using Tab or Shift+Tab.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
